drivers/adc/adc_stm32.c

```diff
@@ -1351,19 +1210,19 @@ static int adc_stm32_init(const struct device *dev)
 	}
 
 	data->dev = dev;
-#if defined(CONFIG_SOC_SERIES_STM32C0X) || \
-	defined(CONFIG_SOC_SERIES_STM32F0X) || \
-	defined(CONFIG_SOC_SERIES_STM32G0X) || \
-	defined(CONFIG_SOC_SERIES_STM32L0X)
-	/*
-	 * All conversion time for all channels on one ADC instance for F0 and
-	 * L0 series chips has to be the same. For C0 and G0 currently only one
-	 * of the two available common channel conversion times is used.
-	 * This additional variable is for checking if the conversion time
-	 * selection of all channels on one ADC instance is the same.
-	 */
-	data->acq_time_index = -1;
-#endif
+
+	if (config->sampling_time_common_channel_nb) {
```

Comment:
Minor optimization would be to omit this if statement.  Setting to -1 when in won't get used later causes no harm, and -1 means "unset" here so it's not really wrong.  It takes just as many instructions to check if `config->sampling_time_common_channel_nb` is non-zero it saves by skipping setting `data->acq_time_index`.

Reply:
Done